File: backend/src/domain/glucose_models.py

from enum import Enum
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict, Field

class GlucoseType(str, Enum):
    FINGER = "FINGER"
    CGM = "CGM"
    MANUAL = "MANUAL"

class GlucoseCreateRequest(BaseModel):
    value: int = Field(..., ge=20, le=600, description="Glucose value in mg/dL")
    timestamp: datetime = Field(default_factory=datetime.utcnow)
    measurement_type: GlucoseType = Field(default=GlucoseType.FINGER)
    notes: Optional[str] = None

from uuid import UUID

class GlucoseResponse(BaseModel):
    id: UUID
    patient_id: UUID
    value: int = Field(validation_alias="glucose_value")
    timestamp: datetime
    measurement_type: GlucoseType
    notes: Optional[str] = None
    
    model_config = ConfigDict(from_attributes=True)

File: frontend/lib/data/models/glucose_models.dart

import 'package:json_annotation/json_annotation.dart';

part 'glucose_models.g.dart';

enum GlucoseType {
  @JsonValue('FINGER')
  finger,
  @JsonValue('CGM')
  cgm,
  @JsonValue('MANUAL')
  manual,
}

@JsonSerializable()
class GlucoseMeasurement {
  final String id;
  @JsonKey(name: 'patient_id')
  final String patientId;
  @JsonKey(name: 'glucose_value')
  final int glucoseValue;
  final DateTime timestamp;
  @JsonKey(name: 'measurement_type')
  final GlucoseType measurementType;
  final String? notes;

  GlucoseMeasurement({
    required this.id,
    required this.patientId,
    required this.glucoseValue,
    required this.timestamp,
    this.measurementType = GlucoseType.finger,
    this.notes,
  });

  factory GlucoseMeasurement.fromJson(Map<String, dynamic> json) =>
      _$GlucoseMeasurementFromJson(json);

  Map<String, dynamic> toJson() => _$GlucoseMeasurementToJson(this);
}

@JsonSerializable()
class GlucoseCreateRequest {
  final int value;
  final DateTime timestamp;
  @JsonKey(name: 'measurement_type')
  final GlucoseType measurementType;
  final String? notes;

  GlucoseCreateRequest({
    required this.value,
    required this.timestamp,
    this.measurementType = GlucoseType.finger,
    this.notes,
  });

  Map<String, dynamic> toJson() => _$GlucoseCreateRequestToJson(this);
}
